packages/backend-app-api/src/services/implementations/httpAuth/httpAuthServiceFactory.ts

```diff
@@ -235,14 +235,24 @@ class DefaultHttpAuthService implements HttpAuthService {
       return undefined;
     }
 
-    const existingCredentials = await this.#auth.authenticate(existingCookie, {
-      allowLimitedAccess: true,
-    });
-    if (!this.#auth.isPrincipal(existingCredentials, 'user')) {
-      return undefined;
-    }
+    try {
+      const existingCredentials = await this.#auth.authenticate(
+        existingCookie,
+        {
+          allowLimitedAccess: true,
+        },
+      );
+      if (!this.#auth.isPrincipal(existingCredentials, 'user')) {
+        return undefined;
+      }
 
-    return existingCredentials.expiresAt;
+      return existingCredentials.expiresAt;
+    } catch (error) {
+      if (error instanceof AuthenticationError) {
```

Comment:
This check isn't safe, use `error.name === 'AuthenticationError'` instead, to be able to handle the case where `@backstage/errors` is duplicated.

Reply:
I'll fix it too.